Define a virtual PlatformMediaSession interface and have PlatformMediaSession derive from it

eric-carlson opened the pull request:
#### 8f86af059bbbe6383dc4a6ee7e37a63e103afa71
<pre>
Define a virtual PlatformMediaSession interface and have PlatformMediaSession derive from it
<a href="https://bugs.webkit.org/show_bug.cgi?id=291374">https://bugs.webkit.org/show_bug.cgi?id=291374</a>
<a href="https://rdar.apple.com/148988705">rdar://148988705</a>

Reviewed by Andy Estes.

* Source/WebCore/Headers.cmake

* Source/WebCore/Modules/webaudio/AudioContext.cpp:
(WebCore::AudioContext::AudioContext):
(WebCore::AudioContext::selectBestMediaSession):
* Source/WebCore/Modules/webaudio/AudioContext.h:

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::initializeMediaSession):
(WebCore::HTMLMediaElement::selectBestMediaSession):
(WebCore::HTMLMediaElement::virtualHasPendingActivity const):
(WebCore::HTMLMediaElement::displayType const):
(WebCore::HTMLMediaElement::protectedLogger const): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::mediaSessionIfExists const):
(WebCore::HTMLMediaElement::isTemporarilyAllowingInlinePlaybackAfterFullscreen const):

* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::MediaElementSession):
* Source/WebCore/html/MediaElementSession.h:
(isType):

* Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
(WebCore::MediaSessionManagerInterface::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerInterface::clientCharacteristicsChanged):

* Source/WebCore/platform/audio/PlatformMediaSession.cpp:
(WebCore::PlatformMediaSession::setState):
(WebCore::PlatformMediaSession::pauseSession):
(WebCore::PlatformMediaSession::stopSession):
(WebCore::PlatformMediaSession::didReceiveRemoteControlCommand):
(WebCore::PlatformMediaSession::canPlayConcurrently const):
(WebCore::PlatformMediaSession::selectBestMediaSession):
(WebCore::PlatformMediaSession::create): Deleted.
(WebCore::PlatformMediaSession::PlatformMediaSession): Deleted.
(WebCore::PlatformMediaSession::mediaType const): Deleted.
(WebCore::PlatformMediaSession::presentationType const): Deleted.
(WebCore::PlatformMediaSession::canReceiveRemoteControlCommands const): Deleted.
(WebCore::PlatformMediaSession::supportsSeeking const): Deleted.
(WebCore::PlatformMediaSession::isSuspended const): Deleted.
(WebCore::PlatformMediaSession::isPlaying const): Deleted.
(WebCore::PlatformMediaSession::isAudible const): Deleted.
(WebCore::PlatformMediaSession::isEnded const): Deleted.
(WebCore::PlatformMediaSession::duration const): Deleted.
(WebCore::PlatformMediaSession::shouldOverrideBackgroundLoadingRestriction const): Deleted.
(WebCore::PlatformMediaSession::displayType const): Deleted.
(WebCore::PlatformMediaSession::canProduceAudio const): Deleted.
(WebCore::PlatformMediaSession::hasMediaStreamSource const): Deleted.
(WebCore::PlatformMediaSession::shouldOverridePauseDuringRouteChange const): Deleted.
(WebCore::PlatformMediaSession::nowPlayingInfo const): Deleted.
(WebCore::PlatformMediaSession::isNowPlayingEligible const): Deleted.
(WebCore::PlatformMediaSession::presentingApplicationPID const): Deleted.
(WebCore::PlatformMediaSessionClient::mediaSessionDuration const): Deleted.
(WebCore::PlatformMediaSessionClient::nowPlayingInfo const): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSession.h:
(WebCore::PlatformMediaSession::create):
(WebCore::PlatformMediaSession::PlatformMediaSession):
(WebCore::PlatformMediaSession::state const): Deleted.
(WebCore::PlatformMediaSession::stateToRestore const): Deleted.
(WebCore::PlatformMediaSession::suspendBuffering): Deleted.
(WebCore::PlatformMediaSession::resumeBuffering): Deleted.
(WebCore::PlatformMediaSession::isPlayingToWirelessPlaybackTarget const): Deleted.
(WebCore::PlatformMediaSession::requiresPlaybackTargetRouteMonitoring const): Deleted.
(WebCore::PlatformMediaSession::resetPlaybackSessionState): Deleted.
(WebCore::PlatformMediaSession::hasPlayedAudiblySinceLastInterruption const): Deleted.
(WebCore::PlatformMediaSession::clearHasPlayedAudiblySinceLastInterruption): Deleted.
(WebCore::PlatformMediaSession::preparingToPlay const): Deleted.
(WebCore::PlatformMediaSession::updateMediaUsageIfChanged): Deleted.
(WebCore::PlatformMediaSession::isLongEnoughForMainContent const): Deleted.
(WebCore::PlatformMediaSession::mediaSessionIdentifier const): Deleted.
(WebCore::PlatformMediaSession::isActiveNowPlayingSession const): Deleted.
(WebCore::PlatformMediaSession::audioSessionCategoryChanged): Deleted.
(WebCore::PlatformMediaSession::client const): Deleted.
(WebCore::PlatformMediaSessionClient::displayType const): Deleted.
(WebCore::PlatformMediaSessionClient::resumeAutoplaying): Deleted.
(WebCore::PlatformMediaSessionClient::canProduceAudio const): Deleted.
(WebCore::PlatformMediaSessionClient::isSuspended const): Deleted.
(WebCore::PlatformMediaSessionClient::isPlaying const): Deleted.
(WebCore::PlatformMediaSessionClient::isAudible const): Deleted.
(WebCore::PlatformMediaSessionClient::isEnded const): Deleted.
(WebCore::PlatformMediaSessionClient::shouldOverrideBackgroundLoadingRestriction const): Deleted.
(WebCore::PlatformMediaSessionClient::wirelessRoutesAvailableDidChange): Deleted.
(WebCore::PlatformMediaSessionClient::setWirelessPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionClient::isPlayingToWirelessPlaybackTarget const): Deleted.
(WebCore::PlatformMediaSessionClient::setShouldPlayToPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionClient::playbackTargetPickerWasDismissed): Deleted.
(WebCore::PlatformMediaSessionClient::isPlayingOnSecondScreen const): Deleted.
(WebCore::PlatformMediaSessionClient::hasMediaStreamSource const): Deleted.
(WebCore::PlatformMediaSessionClient::processIsSuspendedChanged): Deleted.
(WebCore::PlatformMediaSessionClient::shouldOverridePauseDuringRouteChange const): Deleted.
(WebCore::PlatformMediaSessionClient::isNowPlayingEligible const): Deleted.
(WebCore::PlatformMediaSessionClient::selectBestMediaSession): Deleted.

* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
(WebCore::PlatformMediaSessionClient::displayType const):
(WebCore::PlatformMediaSessionClient::resumeAutoplaying):
(WebCore::PlatformMediaSessionClient::canProduceAudio const):
(WebCore::PlatformMediaSessionClient::isSuspended const):
(WebCore::PlatformMediaSessionClient::isPlaying const):
(WebCore::PlatformMediaSessionClient::isAudible const):
(WebCore::PlatformMediaSessionClient::isEnded const):
(WebCore::PlatformMediaSessionClient::mediaSessionDuration const):
(WebCore::PlatformMediaSessionClient::shouldOverrideBackgroundLoadingRestriction const):
(WebCore::PlatformMediaSessionClient::wirelessRoutesAvailableDidChange):
(WebCore::PlatformMediaSessionClient::setWirelessPlaybackTarget):
(WebCore::PlatformMediaSessionClient::isPlayingToWirelessPlaybackTarget const):
(WebCore::PlatformMediaSessionClient::setShouldPlayToPlaybackTarget):
(WebCore::PlatformMediaSessionClient::playbackTargetPickerWasDismissed):
(WebCore::PlatformMediaSessionClient::isPlayingOnSecondScreen const):
(WebCore::PlatformMediaSessionClient::hasMediaStreamSource const):
(WebCore::PlatformMediaSessionClient::processIsSuspendedChanged):
(WebCore::PlatformMediaSessionClient::shouldOverridePauseDuringRouteChange const):
(WebCore::PlatformMediaSessionClient::isNowPlayingEligible const):
(WebCore::PlatformMediaSessionClient::nowPlayingInfo const):
(WebCore::PlatformMediaSessionClient::selectBestMediaSession):
(WebCore::PlatformMediaSessionClient::protectedLogger const):
(WebCore::PlatformMediaSessionInterface::mediaType const):
(WebCore::PlatformMediaSessionInterface::presentationType const):
(WebCore::PlatformMediaSessionInterface::suspendBuffering):
(WebCore::PlatformMediaSessionInterface::resumeBuffering):
(WebCore::PlatformMediaSessionInterface::canReceiveRemoteControlCommands const):
(WebCore::PlatformMediaSessionInterface::displayType const):
(WebCore::PlatformMediaSessionInterface::supportsSeeking const):
(WebCore::PlatformMediaSessionInterface::isSuspended const):
(WebCore::PlatformMediaSessionInterface::isPlaying const):
(WebCore::PlatformMediaSessionInterface::isAudible const):
(WebCore::PlatformMediaSessionInterface::isEnded const):
(WebCore::PlatformMediaSessionInterface::duration const):
(WebCore::PlatformMediaSessionInterface::shouldOverrideBackgroundLoadingRestriction const):
(WebCore::PlatformMediaSessionInterface::isPlayingToWirelessPlaybackTarget const):
(WebCore::PlatformMediaSessionInterface::setPlaybackTarget):
(WebCore::PlatformMediaSessionInterface::externalOutputDeviceAvailableDidChange):
(WebCore::PlatformMediaSessionInterface::setShouldPlayToPlaybackTarget):
(WebCore::PlatformMediaSessionInterface::playbackTargetPickerWasDismissed):
(WebCore::PlatformMediaSessionInterface::requiresPlaybackTargetRouteMonitoring const):
(WebCore::PlatformMediaSessionInterface::canProduceAudio const):
(WebCore::PlatformMediaSessionInterface::hasMediaStreamSource const):
(WebCore::PlatformMediaSessionInterface::resetPlaybackSessionState):
(WebCore::PlatformMediaSessionInterface::hasPlayedAudiblySinceLastInterruption const):
(WebCore::PlatformMediaSessionInterface::setHasPlayedAudiblySinceLastInterruption):
(WebCore::PlatformMediaSessionInterface::shouldOverridePauseDuringRouteChange const):
(WebCore::PlatformMediaSessionInterface::nowPlayingInfo const):
(WebCore::PlatformMediaSessionInterface::isNowPlayingEligible const):
(WebCore::PlatformMediaSessionInterface::updateMediaUsageIfChanged):
(WebCore::PlatformMediaSessionInterface::isLongEnoughForMainContent const):
(WebCore::PlatformMediaSessionInterface::mediaSessionIdentifier const):
(WebCore::PlatformMediaSessionInterface::presentingApplicationPID const):
(WebCore::PlatformMediaSessionInterface::audioSessionCategoryChanged):
(WebCore::PlatformMediaSessionInterface::client const):
(WebCore::PlatformMediaSessionInterface::protectedLogger const):
(WebCore::PlatformMediaSessionInterface::PlatformMediaSessionInterface):

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::addSession):
(WebCore::PlatformMediaSessionManager::removeSession):
(WebCore::PlatformMediaSessionManager::sessionWillBeginPlayback):
(WebCore::PlatformMediaSessionManager::sessionWillEndPlayback):
(WebCore::PlatformMediaSessionManager::sessionStateChanged):
(WebCore::PlatformMediaSessionManager::setCurrentSession):
(WebCore::PlatformMediaSessionManager::currentSession const):
(WebCore::PlatformMediaSessionManager::sessionIsPlayingToWirelessPlaybackTargetChanged):
(WebCore::PlatformMediaSessionManager::computeSupportsSeeking const):
(WebCore::PlatformMediaSessionManager::sessionsMatching const):
(WebCore::PlatformMediaSessionManager::firstSessionMatching const):
(WebCore::PlatformMediaSessionManager::forEachMatchingSession):
(WebCore::PlatformMediaSessionManager::forEachSessionInGroup):
(WebCore::PlatformMediaSessionManager::forEachSession):
(WebCore::PlatformMediaSessionManager::anyOfSessions const):
(WebCore::PlatformMediaSessionManager::bestEligibleSessionForRemoteControls):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:

* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::beginInterruption):
(WebCore::MediaSessionManagerCocoa::sessionWillBeginPlayback):
(WebCore::MediaSessionManagerCocoa::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerCocoa::addSession):
(WebCore::MediaSessionManagerCocoa::removeSession):
(WebCore::MediaSessionManagerCocoa::setCurrentSession):
(WebCore::MediaSessionManagerCocoa::sessionWillEndPlayback):
(WebCore::MediaSessionManagerCocoa::clientCharacteristicsChanged):
(WebCore::MediaSessionManagerCocoa::nowPlayingEligibleSession):
(WebCore::MediaSessionManagerCocoa::updateActiveNowPlayingSession):
(WebCore::MediaSessionManagerCocoa::updateNowPlayingInfo):

* Source/WebCore/platform/audio/glib/MediaSessionGLib.cpp:
(WebCore::MediaSessionGLib::getPlaybackStatusAsGVariant):
(WebCore::MediaSessionGLib::playbackStatusChanged):
* Source/WebCore/platform/audio/glib/MediaSessionGLib.h:

* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.cpp:
(WebCore::MediaSessionManagerGLib::sessionWillBeginPlayback):
(WebCore::MediaSessionManagerGLib::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerGLib::addSession):
(WebCore::MediaSessionManagerGLib::removeSession):
(WebCore::MediaSessionManagerGLib::setCurrentSession):
(WebCore::MediaSessionManagerGLib::sessionWillEndPlayback):
(WebCore::MediaSessionManagerGLib::sessionStateChanged):
(WebCore::MediaSessionManagerGLib::clientCharacteristicsChanged):
(WebCore::MediaSessionManagerGLib::setPrimarySessionIfNeeded):
(WebCore::MediaSessionManagerGLib::unregisterAllOtherSessions):
* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.h:
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.mm:
(WebCore::MediaSessionManageriOS::sessionWillBeginPlayback):
(WebCore::MediaSessionManageriOS::sessionWillEndPlayback):
(WebCore::MediaSessionManageriOS::activeVideoRouteDidChange):

* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
(WebCore::VideoPresentationModelVideoElement::videoInteractedWith):

Canonical link: <a href="https://commits.webkit.org/293547@main">https://commits.webkit.org/293547@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/20591988f3caa207939a48393666ce4745ddde32

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99232 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18879 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9126 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104361 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/49829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101272 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19168 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27313 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75533 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/49829 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102239 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14567 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89600 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55894 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14359 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7578 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49191 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84290 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/7665 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106718 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26345 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/19204 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84495 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26708 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85801 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84011 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21309 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28663 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6341 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20070 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26285 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/31470 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26106 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29419 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27673 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aestes commented on Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
```diff
@@ -0,0 +1,331 @@
+class PlatformMediaSessionManager;
+enum class AudioSessionCategory : uint8_t;
+enum class AudioSessionMode : uint8_t;
+enum class DelayCallingUpdateNowPlaying : bool { No, Yes };
```
Nit: I would move this down to where other enum classes are defined rather than commingle it with forward declarations.

eric-carlson committed:
Define a virtual PlatformMediaSession interface and have PlatformMediaSession derive from it
https://bugs.webkit.org/show_bug.cgi?id=291374
rdar://148988705

Reviewed by Andy Estes.

* Source/WebCore/Headers.cmake

* Source/WebCore/Modules/webaudio/AudioContext.cpp:
(WebCore::AudioContext::AudioContext):
(WebCore::AudioContext::selectBestMediaSession):
* Source/WebCore/Modules/webaudio/AudioContext.h:

* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::initializeMediaSession):
(WebCore::HTMLMediaElement::selectBestMediaSession):
(WebCore::HTMLMediaElement::virtualHasPendingActivity const):
(WebCore::HTMLMediaElement::displayType const):
(WebCore::HTMLMediaElement::protectedLogger const): Deleted.
* Source/WebCore/html/HTMLMediaElement.h:
(WebCore::HTMLMediaElement::mediaSessionIfExists const):
(WebCore::HTMLMediaElement::isTemporarilyAllowingInlinePlaybackAfterFullscreen const):

* Source/WebCore/html/MediaElementSession.cpp:
(WebCore::MediaElementSession::MediaElementSession):
* Source/WebCore/html/MediaElementSession.h:
(isType):

* Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
(WebCore::MediaSessionManagerInterface::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerInterface::clientCharacteristicsChanged):

* Source/WebCore/platform/audio/PlatformMediaSession.cpp:
(WebCore::PlatformMediaSession::setState):
(WebCore::PlatformMediaSession::pauseSession):
(WebCore::PlatformMediaSession::stopSession):
(WebCore::PlatformMediaSession::didReceiveRemoteControlCommand):
(WebCore::PlatformMediaSession::canPlayConcurrently const):
(WebCore::PlatformMediaSession::selectBestMediaSession):
(WebCore::PlatformMediaSession::create): Deleted.
(WebCore::PlatformMediaSession::PlatformMediaSession): Deleted.
(WebCore::PlatformMediaSession::mediaType const): Deleted.
(WebCore::PlatformMediaSession::presentationType const): Deleted.
(WebCore::PlatformMediaSession::canReceiveRemoteControlCommands const): Deleted.
(WebCore::PlatformMediaSession::supportsSeeking const): Deleted.
(WebCore::PlatformMediaSession::isSuspended const): Deleted.
(WebCore::PlatformMediaSession::isPlaying const): Deleted.
(WebCore::PlatformMediaSession::isAudible const): Deleted.
(WebCore::PlatformMediaSession::isEnded const): Deleted.
(WebCore::PlatformMediaSession::duration const): Deleted.
(WebCore::PlatformMediaSession::shouldOverrideBackgroundLoadingRestriction const): Deleted.
(WebCore::PlatformMediaSession::displayType const): Deleted.
(WebCore::PlatformMediaSession::canProduceAudio const): Deleted.
(WebCore::PlatformMediaSession::hasMediaStreamSource const): Deleted.
(WebCore::PlatformMediaSession::shouldOverridePauseDuringRouteChange const): Deleted.
(WebCore::PlatformMediaSession::nowPlayingInfo const): Deleted.
(WebCore::PlatformMediaSession::isNowPlayingEligible const): Deleted.
(WebCore::PlatformMediaSession::presentingApplicationPID const): Deleted.
(WebCore::PlatformMediaSessionClient::mediaSessionDuration const): Deleted.
(WebCore::PlatformMediaSessionClient::nowPlayingInfo const): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSession.h:
(WebCore::PlatformMediaSession::create):
(WebCore::PlatformMediaSession::PlatformMediaSession):
(WebCore::PlatformMediaSession::state const): Deleted.
(WebCore::PlatformMediaSession::stateToRestore const): Deleted.
(WebCore::PlatformMediaSession::suspendBuffering): Deleted.
(WebCore::PlatformMediaSession::resumeBuffering): Deleted.
(WebCore::PlatformMediaSession::isPlayingToWirelessPlaybackTarget const): Deleted.
(WebCore::PlatformMediaSession::requiresPlaybackTargetRouteMonitoring const): Deleted.
(WebCore::PlatformMediaSession::resetPlaybackSessionState): Deleted.
(WebCore::PlatformMediaSession::hasPlayedAudiblySinceLastInterruption const): Deleted.
(WebCore::PlatformMediaSession::clearHasPlayedAudiblySinceLastInterruption): Deleted.
(WebCore::PlatformMediaSession::preparingToPlay const): Deleted.
(WebCore::PlatformMediaSession::updateMediaUsageIfChanged): Deleted.
(WebCore::PlatformMediaSession::isLongEnoughForMainContent const): Deleted.
(WebCore::PlatformMediaSession::mediaSessionIdentifier const): Deleted.
(WebCore::PlatformMediaSession::isActiveNowPlayingSession const): Deleted.
(WebCore::PlatformMediaSession::audioSessionCategoryChanged): Deleted.
(WebCore::PlatformMediaSession::client const): Deleted.
(WebCore::PlatformMediaSessionClient::displayType const): Deleted.
(WebCore::PlatformMediaSessionClient::resumeAutoplaying): Deleted.
(WebCore::PlatformMediaSessionClient::canProduceAudio const): Deleted.
(WebCore::PlatformMediaSessionClient::isSuspended const): Deleted.
(WebCore::PlatformMediaSessionClient::isPlaying const): Deleted.
(WebCore::PlatformMediaSessionClient::isAudible const): Deleted.
(WebCore::PlatformMediaSessionClient::isEnded const): Deleted.
(WebCore::PlatformMediaSessionClient::shouldOverrideBackgroundLoadingRestriction const): Deleted.
(WebCore::PlatformMediaSessionClient::wirelessRoutesAvailableDidChange): Deleted.
(WebCore::PlatformMediaSessionClient::setWirelessPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionClient::isPlayingToWirelessPlaybackTarget const): Deleted.
(WebCore::PlatformMediaSessionClient::setShouldPlayToPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionClient::playbackTargetPickerWasDismissed): Deleted.
(WebCore::PlatformMediaSessionClient::isPlayingOnSecondScreen const): Deleted.
(WebCore::PlatformMediaSessionClient::hasMediaStreamSource const): Deleted.
(WebCore::PlatformMediaSessionClient::processIsSuspendedChanged): Deleted.
(WebCore::PlatformMediaSessionClient::shouldOverridePauseDuringRouteChange const): Deleted.
(WebCore::PlatformMediaSessionClient::isNowPlayingEligible const): Deleted.
(WebCore::PlatformMediaSessionClient::selectBestMediaSession): Deleted.

* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
(WebCore::PlatformMediaSessionClient::displayType const):
(WebCore::PlatformMediaSessionClient::resumeAutoplaying):
(WebCore::PlatformMediaSessionClient::canProduceAudio const):
(WebCore::PlatformMediaSessionClient::isSuspended const):
(WebCore::PlatformMediaSessionClient::isPlaying const):
(WebCore::PlatformMediaSessionClient::isAudible const):
(WebCore::PlatformMediaSessionClient::isEnded const):
(WebCore::PlatformMediaSessionClient::mediaSessionDuration const):
(WebCore::PlatformMediaSessionClient::shouldOverrideBackgroundLoadingRestriction const):
(WebCore::PlatformMediaSessionClient::wirelessRoutesAvailableDidChange):
(WebCore::PlatformMediaSessionClient::setWirelessPlaybackTarget):
(WebCore::PlatformMediaSessionClient::isPlayingToWirelessPlaybackTarget const):
(WebCore::PlatformMediaSessionClient::setShouldPlayToPlaybackTarget):
(WebCore::PlatformMediaSessionClient::playbackTargetPickerWasDismissed):
(WebCore::PlatformMediaSessionClient::isPlayingOnSecondScreen const):
(WebCore::PlatformMediaSessionClient::hasMediaStreamSource const):
(WebCore::PlatformMediaSessionClient::processIsSuspendedChanged):
(WebCore::PlatformMediaSessionClient::shouldOverridePauseDuringRouteChange const):
(WebCore::PlatformMediaSessionClient::isNowPlayingEligible const):
(WebCore::PlatformMediaSessionClient::nowPlayingInfo const):
(WebCore::PlatformMediaSessionClient::selectBestMediaSession):
(WebCore::PlatformMediaSessionClient::protectedLogger const):
(WebCore::PlatformMediaSessionInterface::mediaType const):
(WebCore::PlatformMediaSessionInterface::presentationType const):
(WebCore::PlatformMediaSessionInterface::suspendBuffering):
(WebCore::PlatformMediaSessionInterface::resumeBuffering):
(WebCore::PlatformMediaSessionInterface::canReceiveRemoteControlCommands const):
(WebCore::PlatformMediaSessionInterface::displayType const):
(WebCore::PlatformMediaSessionInterface::supportsSeeking const):
(WebCore::PlatformMediaSessionInterface::isSuspended const):
(WebCore::PlatformMediaSessionInterface::isPlaying const):
(WebCore::PlatformMediaSessionInterface::isAudible const):
(WebCore::PlatformMediaSessionInterface::isEnded const):
(WebCore::PlatformMediaSessionInterface::duration const):
(WebCore::PlatformMediaSessionInterface::shouldOverrideBackgroundLoadingRestriction const):
(WebCore::PlatformMediaSessionInterface::isPlayingToWirelessPlaybackTarget const):
(WebCore::PlatformMediaSessionInterface::setPlaybackTarget):
(WebCore::PlatformMediaSessionInterface::externalOutputDeviceAvailableDidChange):
(WebCore::PlatformMediaSessionInterface::setShouldPlayToPlaybackTarget):
(WebCore::PlatformMediaSessionInterface::playbackTargetPickerWasDismissed):
(WebCore::PlatformMediaSessionInterface::requiresPlaybackTargetRouteMonitoring const):
(WebCore::PlatformMediaSessionInterface::canProduceAudio const):
(WebCore::PlatformMediaSessionInterface::hasMediaStreamSource const):
(WebCore::PlatformMediaSessionInterface::resetPlaybackSessionState):
(WebCore::PlatformMediaSessionInterface::hasPlayedAudiblySinceLastInterruption const):
(WebCore::PlatformMediaSessionInterface::setHasPlayedAudiblySinceLastInterruption):
(WebCore::PlatformMediaSessionInterface::shouldOverridePauseDuringRouteChange const):
(WebCore::PlatformMediaSessionInterface::nowPlayingInfo const):
(WebCore::PlatformMediaSessionInterface::isNowPlayingEligible const):
(WebCore::PlatformMediaSessionInterface::updateMediaUsageIfChanged):
(WebCore::PlatformMediaSessionInterface::isLongEnoughForMainContent const):
(WebCore::PlatformMediaSessionInterface::mediaSessionIdentifier const):
(WebCore::PlatformMediaSessionInterface::presentingApplicationPID const):
(WebCore::PlatformMediaSessionInterface::audioSessionCategoryChanged):
(WebCore::PlatformMediaSessionInterface::client const):
(WebCore::PlatformMediaSessionInterface::protectedLogger const):
(WebCore::PlatformMediaSessionInterface::PlatformMediaSessionInterface):

* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::addSession):
(WebCore::PlatformMediaSessionManager::removeSession):
(WebCore::PlatformMediaSessionManager::sessionWillBeginPlayback):
(WebCore::PlatformMediaSessionManager::sessionWillEndPlayback):
(WebCore::PlatformMediaSessionManager::sessionStateChanged):
(WebCore::PlatformMediaSessionManager::setCurrentSession):
(WebCore::PlatformMediaSessionManager::currentSession const):
(WebCore::PlatformMediaSessionManager::sessionIsPlayingToWirelessPlaybackTargetChanged):
(WebCore::PlatformMediaSessionManager::computeSupportsSeeking const):
(WebCore::PlatformMediaSessionManager::sessionsMatching const):
(WebCore::PlatformMediaSessionManager::firstSessionMatching const):
(WebCore::PlatformMediaSessionManager::forEachMatchingSession):
(WebCore::PlatformMediaSessionManager::forEachSessionInGroup):
(WebCore::PlatformMediaSessionManager::forEachSession):
(WebCore::PlatformMediaSessionManager::anyOfSessions const):
(WebCore::PlatformMediaSessionManager::bestEligibleSessionForRemoteControls):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:

* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::beginInterruption):
(WebCore::MediaSessionManagerCocoa::sessionWillBeginPlayback):
(WebCore::MediaSessionManagerCocoa::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerCocoa::addSession):
(WebCore::MediaSessionManagerCocoa::removeSession):
(WebCore::MediaSessionManagerCocoa::setCurrentSession):
(WebCore::MediaSessionManagerCocoa::sessionWillEndPlayback):
(WebCore::MediaSessionManagerCocoa::clientCharacteristicsChanged):
(WebCore::MediaSessionManagerCocoa::nowPlayingEligibleSession):
(WebCore::MediaSessionManagerCocoa::updateActiveNowPlayingSession):
(WebCore::MediaSessionManagerCocoa::updateNowPlayingInfo):

* Source/WebCore/platform/audio/glib/MediaSessionGLib.cpp:
(WebCore::MediaSessionGLib::getPlaybackStatusAsGVariant):
(WebCore::MediaSessionGLib::playbackStatusChanged):
* Source/WebCore/platform/audio/glib/MediaSessionGLib.h:

* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.cpp:
(WebCore::MediaSessionManagerGLib::sessionWillBeginPlayback):
(WebCore::MediaSessionManagerGLib::sessionDidEndRemoteScrubbing):
(WebCore::MediaSessionManagerGLib::addSession):
(WebCore::MediaSessionManagerGLib::removeSession):
(WebCore::MediaSessionManagerGLib::setCurrentSession):
(WebCore::MediaSessionManagerGLib::sessionWillEndPlayback):
(WebCore::MediaSessionManagerGLib::sessionStateChanged):
(WebCore::MediaSessionManagerGLib::clientCharacteristicsChanged):
(WebCore::MediaSessionManagerGLib::setPrimarySessionIfNeeded):
(WebCore::MediaSessionManagerGLib::unregisterAllOtherSessions):
* Source/WebCore/platform/audio/glib/MediaSessionManagerGLib.h:
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.mm:
(WebCore::MediaSessionManageriOS::sessionWillBeginPlayback):
(WebCore::MediaSessionManageriOS::sessionWillEndPlayback):
(WebCore::MediaSessionManageriOS::activeVideoRouteDidChange):

* Source/WebCore/platform/cocoa/VideoPresentationModelVideoElement.mm:
(WebCore::VideoPresentationModelVideoElement::videoInteractedWith):

Canonical link: https://commits.webkit.org/293547@main
